WordPress/gutenberg
# Navigation screen: Atomic save using customizer API endpoint

## Description

This PR enables saving the entire navigation using a single HTTP request.

The current behavior is sending one request per change at the moment. If you e.g. add 20 menu items, update 17 and remove 10, it will result in 47 http requests. This is pretty fragile and would easily break under a number of scenarios - validation errors, network failures, accidentally closing the browser, and so on.

This PR proposes the following implementation:
* Whenever a new navigation link block is added, we create a draft menu item post - this is consistent with the way existing menus page works.
* Save action sends the entire state over to the existing customizer endpoint where it's validated in bulk and then saved.

The previous approach also proposed a new batch endpoint for handling these requests - ultimately we agreed on implementing this on the core level instead of adding a custom endpoint:  https://github.com/WordPress/gutenberg/pull/22148 

## How has this been tested?
1. Enable the navigation experiment in Gutenberg > Experiments
1. Go to Gutenberg > Navigation (beta)
1. Add a few menu items, update a few other, delete a few existing ones, then hit save
1. Confirm everything was saved

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
